### PR TITLE
Fix 69: iOS 7 shifting focus points due to address bar showing

### PR DESCRIPTION
**Pull request by Helen-Mobify**
We found a way to fix the shifted focus points when it happened. We need to do focus and then blur on an input element. However, on iOS 7.x, the resize event does not fire when the address bar/footer shows or hides.

Reference: https://steveush.wordpress.com/2014/03/13/safari-on-ios7-address-and-bookmark-bars-fixed-position-bug/

We are thinking to attempt the solution above but only poll the window height while pinny is open. https://gist.github.com/scalvert/1498b1e8a4e5ca89df17
 
Status: **Opened for visibility**

Reviewers: @scalvert 
JIRA: https://github.com/mobify/pinny/issues/69

How to test drive this PR:
1. Use a device running iOS 7.0.x
2. `grunt serve`
3. Navigate to the examples 
4. Partially scroll the page
5. Open a pinny
6. Tap the bottom ~50px to trigger the bottom status bar
7. Ensure pinny closes, and/or input fields have their cursor correctly positioned

### CODE REVIEW

**Comment by Helen-Mobify** [comment]
@scalvert 

I have uploaded what I have here. These codes obviously need some major clean up and be put in the right place. Let's figure out how we should approach from here. 

Very interesting find here. We thought that we need to do a focus/blur trick so that the focus points will restore itself. However, the following line in the refresh function will do exactly what we need.
`window.scrollTo(document.body.scrollLeft, document.body.scrollTop);`

**Comment by Helen-Mobify** [comment]
Documenting what @scalvert and I discussed offline.

We are going to move the positionFix file out and make it into a plugin fix.

https://github.com/mobify/address-bar-monitor
(We are still thinking if this is the right name to go with)

The only purpose of this plugin will be firing off custom event when the address bar shows/hides itself. Pinny will listen to these events and perform the required actions in order to fix the focus issue.

The reason why we are moving this file out as a plugin is because this is an issue that will happen to any position fixed elements on iOS 7.

**Comment by Helen-Mobify** [comment]
@scalvert Please take a look and tell me what you think.

File: src/js/pinny.js [hunk 1]
```diff
@@ -11,9 +12,9 @@
         ], factory);
     } else {
         var framework = window.Zepto || window.jQuery;
-        factory(framework, window.Plugin, window.bouncefix);
+        factory(framework, window.Plugin, window.bouncefix, window.syntheticresize);
```
Comment (scalvert) [inline src/js/pinny.js]: `syntheticresize` is never actually bound to window, so this will be undefined.

File: src/js/pinny.js [hunk 2]
```diff
@@ -102,6 +104,8 @@
                 // open and locked the viewport up already
                 !this._activePinnies() && this.$pinny.lockup('lock');
 
+                SyntheticResize.startWatching();
```
Comment (scalvert) [inline src/js/pinny.js]: Correct me if I'm wrong, but I thought this was supposed to only monitor if it was iOS 7. Wouldn't we do that check here, and invoke a callback if it fired? Also, didn't we decide we would call this function `on`, and pass in the resize event and callback?

File: src/js/pinny.js [hunk 3]
```diff
@@ -153,6 +158,9 @@
             this.$pinny.shade('destroy');
             this.$pinny.remove();
 
+            SyntheticResize.destroy();
+            $(window).off(events.resize);
```
Comment (scalvert) [inline src/js/pinny.js]: So we're basically doing both of these things here? Do we want to monitor for resize AND do the synthetic resize polling? If so, this should be behind a separate function that hides this stuff.

File: src/js/pinny.js [hunk 4]
```diff
@@ -195,12 +203,22 @@
...
+
             // Block scrolling on anything but pinny content
-            this.$pinny.on('touchmove', function(e) {
+            container.on('touchmove', function(e) {
                 if (!$(e.target).parents().hasClass(classes.CONTENT)) {
                     e.preventDefault();
                 }
             });
+
+            if ($.os.ios && $.os.major === 7) {
```
Comment (scalvert) [inline src/js/pinny.js]: I'm really confused here. I thought we were doing synthetic resize if this was iOS 7. Is this not the case?

File: src/js/pinny.js (hunk 2 above)
Comment (Helen-Mobify) [inline src/js/pinny.js]: Ah yes! That totally slip my mind.

**Comment by scalvert** [comment]
@kpeatt could you possibly give this one a little look see?

**Comment by kpeatt** [comment]
Seems good to me.

**Pull request merged**